manifests/charts/istio-control/istio-discovery/values.yaml

```diff
@@ -27,9 +27,8 @@ defaults:
   seccompProfile: {}
 
   # Whether to use an existing CNI installation
-  cni:
-    enabled: false
-    provider: default
+  pilot:
```

Comment:
In theory we should continue to respect the previous `pilot.cni.enabled` path as well as the new `pilot.cniEnabled` path for a few releases.

In practice, I would be in favor of just plain dropping the old one - the `enabled` flag here is not the primary mechanism for configuring the inclusion of CNI or anything - it's mostly just a chore-flag you have to keep in sync with the actual install state of `istio-cni`, and as mentioned if we used wrapper charts properly, it could be automatically set in any case.


I'll let others weigh in tho, I'm always in favor of amputating first and seeing who screams and so might not be the best person to ask.

Reply:
Adding deprecated comments to `pilot.cni.enabled` and `pilot.cni.provider` if there is no strong opinion on deleting directly

Reply:
I don't want to keep all 3 - current , deprecated, and extra-deprecated - we should never have more than 1 "deprecated" option.

But if we remove `istio_cni` here we should be good on that front.